pom.xml

```diff
@@ -37,22 +37,27 @@
         <tag>HEAD</tag>
     </scm>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.junit</groupId>
+                <artifactId>junit-bom</artifactId>
+                <version>5.7.0-M1</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
     <dependencies>
         <dependency>
             <groupId>org.hamcrest</groupId>
-            <artifactId>hamcrest-core</artifactId>
-            <version>1.3</version>
+            <artifactId>hamcrest</artifactId>
+            <version>2.2</version>
```

Comment:
I'm happy to bump up JUnit to 5, but I _think_ bumping hamcrest from 1.x to 2.x is backwards incompatible.

So this is probably a major version bump for this library if this change makes it in. 🤔

Reply:
Well, strictly speaking, yes. Though the only parts of Hamcrest used by java-8-matchers are:
- org.hamcrest.Matchers
- org.hamcrest.Description
- org.hamcrest.TypeSafeDiagnosingMatcher
- org.hamcrest.TypeSafeMatcher
<img width="467" alt="Screenshot 2020-07-28 at 22 43 00" src="https://user-images.githubusercontent.com/174823/88719792-1fc12180-d124-11ea-9a74-21f3a9a41644.png">

These classes are already in Hamcrest 1.3, and the APIs of these have not changed in 2.2. The upgrade to Hamcrest 2.2  did not result in any changes in `src/main/java`. If anybody uses java-8-matchers with Hamcrest 1.3 (typically depends directly on Hamcrest 1.3 _as well as_ java-8-matchers), they will work together.

Though there are no mechanisms to tell us if backwards compatibility is broken at a later time, with changes in the library down the road. So the most _correct_ thing may be to bump the major version. But merging this PR will actually _not_ introduce any incompatibility with Hamcrest 1.3.

Reply:
Myself, I really don't have any preference on bumping major version or not, as this is a type of library which is probably mostly, or maybe even always, used as a direct test-dependency, and not pulled in as a transitive dependency of something else. So upgrading `java-8-matchers` is just bumping the version from 1.x to 2, and I don't see the potential of any major version convergence conflicts with other dependencies which themselves may depend on 1.x of `java-8-matchers`.

As I see it, there should be no big deal to bumping the major version for a release after merging this PR.

Reply:
I'll probably make a major version bump then, so at least v1 = Hamcrest 1.x, v2 = Hamcrest 2.x

Reply:
How about merging #6 and #7 first to create a 1.9 release?